packages/strapi-generate-new/lib/before.js

```diff
@@ -45,8 +48,246 @@ module.exports = (scope, cb) => {
+          }
+        }
+      ])
+      .then(answers => {
+        scope.client = answers.client;
+        _.assign(scope.database, {
+          connector: answers.client.connector,
+          settings: {
+            client: answers.client.database
+          },
+          options: {}
+        });
+
+        resolve();
+      });
+    });
+
+    formSteps = formSteps.then(() => {
+      const asyncFn = [
+        new Promise(resolve => {
+          inquirer
+          .prompt([
+            {
+              type: 'input',
+              prefix: '',
+              name: 'name',
+              message: 'Database name:',
+              default: _.get(scope.database, 'database', 'strapi')
+            },
+            {
+              type: 'input',
+              prefix: '',
+              name: 'host',
+              message: 'Host:',
+              default: _.get(scope.database, 'host', 'localhost')
+            },
+            {
+              type: 'input',
+              prefix: '',
+              name: 'port',
+              message: 'Port:',
+              default: (answers) => {
+                if (_.get(scope.database, 'port')) {
+                  return scope.database.port;
+                }
+
+                const ports = {
+                  mongo: 27017,
+                  postgres: 5432,
+                  mysql: 3306,
+                  sqlite3: 1433,
+                  redis: 6379
+                };
+
+                return ports[scope.client.database];
+              }
+            },
+            {
+              type: 'input',
+              prefix: '',
+              name: 'username',
+              message: 'Username:',
+              default: _.get(scope.database, 'username', undefined)
+            },
+            {
+              type: 'input',
+              prefix: '',
+              name: 'password',
+              message: 'Password:',
+              default: _.get(scope.database, 'password', undefined)
+            }
+          ])
+          .then(answers => {
+            scope.database.host = answers.host;
+            scope.database.port = answers.port;
+            scope.database.database = answers.name;
+            scope.database.username = answers.username;
+            scope.database.password = answers.password;
+
+            resolve();
+          });
+        }),
+        new Promise(resolve => {
+          let cmd = `npm install --prefix ${scope.rootPath}_ ${scope.client.connector}@alpha`;
+          if (scope.client.module) {
+            cmd += ` ${scope.client.module}`;
+          }
+
+          exec(cmd, () => {
+            if (scope.client.module) {
+              const lock = require(`${scope.rootPath}_/node_modules/${scope.client.module}/package.json`);
+              scope.client.version = lock.version;
+            }
+
+            resolve();
+          });
+        })
+      ];
+
+      Promise.all(asyncFn)
+      .then(() => {
+        if (scope.client.connector === 'strapi-bookshelf') {
```

Comment:
You should use switch, it's more readable where there is so much code between conditions.

Reply:
I'm also thinking, it could be a great idea to put the code for each connector in the connector itself. It will be more dynamic to add new databases in the future (if we want to).

```js
Promise.all(asyncFn)
.then(() => {
  try {
    // Run the code from the connector itself.
    require(`./node_modules/${scope.client.connector}/utils/connectivity.js`)();
  } catch (e) {
    execSync(`rm -r ${scope.rootPath}_`);
    logger.info('Copying the dashboard...');
  }

  cb.success();
});
```